File: src/dhtdata.fs
module DHTData

open Buffer

type Node =
  { id : Buffer
  ; port : int
  ; host : string
  ; referer : Buffer
  ; distance : Buffer
  ; queried : bool
  ; vectorClock : int
  }
and NodeIdent =
  { id : Buffer
  ; port : int
  ; host : string
  }
and HostIdent =
  { port : int
  ; host : string
  }

let identOfNode (node : Node) =
  { NodeIdent.id = node.id
  ; NodeIdent.host = node.host
  ; NodeIdent.port = node.port
  }

File: src/DHTData.ml
open Buffer

type node =
  { id : string
  ; port : int
  ; host : string
  ; referer : string
  ; distance : string
  ; queried : bool
  ; vectorClock : int
  }
and nodeIdent =
  { id : string
  ; port : int
  ; host : string
  }
and hostIdent =
  { port : int
  ; host : string
  }

let identOfNode (node : node) =
  { id = node.id
  ; host = node.host
  ; port = node.port
  }
